mailpoet/lib/Newsletter/Renderer/Renderer.php

```diff
@@ -83,7 +83,24 @@ private function _render(NewsletterEntity $newsletter, ?SendingQueueEntity $send
     $wpPostEntity = $newsletter->getWpPost();
     $wpPost = $wpPostEntity ? $wpPostEntity->getWpPostInstance() : null;
     if ($wpPost instanceof \WP_Post) {
+      $filterCallback = function($context) use ($newsletter, $sendingQueue): array {
+        if ($sendingQueue) {
+          $task = $sendingQueue->getTask();
+          $subscribers = $task ? $task->getSubscribers() : null;
+          $subscriber = ($subscribers && $subscriber = $subscribers->first()) ? $subscriber->getSubscriber() : null;
+
+          if ($subscriber) {
+            $context['recipient_email'] = $subscriber->getEmail();
+          }
+        }
+        $context['email_type'] = $newsletter->getType();
```

Comment:
I think this is a bit dangerous to do this way. In the renderer, we have no idea who the real recipient is, and we assume it is the first subscriber linked to the task.
This will work for automation emails, which have only one subscriber, but it will create incorrect context data for Newsletter emails, which have multiple subscribers.